src/lib/snark_work_lib/combined_result.mli

```diff
@@ -1,5 +1,5 @@
 type t =
-  { spec_with_proof : (Single_spec.t * Ledger_proof.t) One_or_two.t
+  { results : Single_result.Stable.Latest.t One_or_two.t
```

Comment:
Ah, I see, we're forced to use `Single_spec.Stable.Latest.t` inside the type, although we would be fine with `Single_spec.t`, and avoid some conversions.


What about this?
```suggestion
  { results : (Single_spec.t, Ledger_proof.t) Single_result.Poly.t One_or_two.t
```

Then all newly introduced `read_all_proofs_from_disk` in this PR will go away